utils/docker-utils.sh

```diff
@@ -151,25 +153,57 @@ docker::service_destroy() {
     done
 }
 
+# Removes the stack and awaits for each service in the stack to be removed
+#
+# Arguments:
+# - $1 : stack name to be removed
+# - $@ : service names list (eg. analytics-datastore-elastic-search)
+#
+docker::stack_destroy() {
+    local -r STACK_NAME=${1:?$(missing_param "stack_destroy")}
+    shift
+    log info "Waiting for stack $STACK_NAME to be removed ..."
+    try "docker stack rm \
+        $STACK_NAME" \
+        throw \
+        "Failed to remove $STACK_NAME"
+    overwrite "Waiting for stack $STACK_NAME to be removed ... Done"
+    
+    for service_name in "$@"; do
+        log info "Waiting for service $service_name to be removed ... "
+        if [[ -n $(docker service ls -qf name="${STACK_NAME}"_"${service_name}") ]]; then
+            docker::await_service_destroy "${service_name}" "${STACK_NAME}"
+        fi
+        overwrite "Waiting for service $service_name to be removed ... Done"
+    done
+}
+
 # Tries to remove volumes and retries until it works with a timeout
 #
 # Arguments:
+# - $1 : (optional) stack name that the service falls under (requires the prefix stack=, eg. stack=openhim) (defaults to instant)
 # - $@ : volumes names (eg. es-data psql-1)
 #
 docker::try_remove_volume() {
+    local STACK_NAME="instant"
+    if [[ $1 == stack=* ]]; then
+        STACK_NAME=${1#stack=}
+        shift
+    fi
```

Comment:
The way we are passing the stack name here is different. Is there a specific reason why the prefix "stack=" is used here? I think the format for that arg should be the same throughout, for consistency in the codebase

Reply:
That was initially done as a way to keep backwards compatibility with the instant stack name, since an n-array of volumes could be passed in to maintain backwards compatibility I needed to prefix the first parameter to see if it was a stack name or just another volume name. Hence the strange way setting the stack name. 

However since I now simply prune all volumes that have no services connected to them on stack_destroy, this method has actually been removed from docker-utils